### ISSUE TITLE
Question about the docid generation score

### ISSUE THREAD

**Issue by LouisDo2108** [HansiZeng/PAG#2]
Hi @HansiZeng ,

Could you please kindly point me to the code where you calculate the docid generation score by summing the dot product between hidden states and corresponding token embedding (Equation 2 of your paper)? I cannot seem to find it.

Thank you.

**Comment by HansiZeng** [comment]
I mainly copy and paste the code from the Huggingface (v4.17.0) [generation module](https://github.com/huggingface/transformers/blob/v4.17.0/src/transformers/generation_utils.py#L1855). [My constrained beam search](https://github.com/HansiZeng/PAG/blob/main/t5_pretrainer/tasks/generation.py#L35) (in line 248) is directly copied from their beam search function in line 1855. 
The DocID score is calculated on line 450 of my [beam search](https://github.com/HansiZeng/PAG/blob/main/t5_pretrainer/tasks/generation.py#L35). The original beam search from Huggingface code applies the log_softmax() on the logits, but as you saw in the Eq2 of my paper, I only need the raw logits, hence I comment out the corresponding lines for log_softmax (line 447-449).

B.T.W, to make the code work, please use Huggingface v4.17.0, the latest version has several changes.

**Comment by LouisDo2108** [comment]
Thank you for your quick reply. So the token embedding table here is actually referring to the lm_head linear layer, right?

**Comment by HansiZeng** [comment]
Yes
